### PR TITLE
feat: support rw_depend in sql backend and add sink into dependency info

### LINKED ISSUE #15482: Single node tests failing  `e2e_test/visibility_mode/../batch/./catalog/rw_depend.slt.part`
Screenshot 2024-03-06 at 2 36 15 PM

https://buildkite.com/risingwavelabs/main-cron/builds/1968#018e0de1-7e1a-4451-b233-80e49b58f083

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve #15482 . This PR added index and sink (into table) dependencies, together with some tests. Also changed the system catalog ids to make it compatible with sql backend.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by wangrunji0408 — COMMENTED** [review]
Generally LGTM

**Review by fuyufjh — APPROVED** [review]

File: src/frontend/src/catalog/system_catalog/mod.rs [hunk 1]
```diff
@@ -292,7 +292,7 @@ fn get_acl_items(
 }
 
 pub struct SystemCatalog {
-    // table id = index + 1
+    // table id = index + SYS_CATALOG_START_ID
```
Comment (fuyufjh) [inline src/frontend/src/catalog/system_catalog/mod.rs:295]: A bit weird to me, but acceptable

Comment (yezizp2012) [reply src/frontend/src/catalog/system_catalog/mod.rs:295]: In order to be compatible with sql backend.

File: src/meta/src/manager/catalog/mod.rs [hunk 2]
```diff
@@ -3641,6 +3642,39 @@ impl CatalogManager {
             .collect_vec()
     }
 
+    pub async fn list_object_dependencies(&self) -> Vec<PbObjectDependencies> {
+        let core = &self.core.lock().await.database;
+        let mut dependencies = vec![];
+        for table in core.tables.values() {
```
[the remark this answers is not in the document]
Comment (yezizp2012) [reply src/meta/src/manager/catalog/mod.rs:3648]: Already taking into consideration. They're included in object dependency table.

**Pull request merged**